app/models/response.rb

```diff
@@ -11,10 +11,30 @@
 #  updated_at                           :datetime         not null
 #
 
+class AnswerValidator < ActiveModel::Validator
+  def validate(record)
+    if record.question.question_type != 'short_answer'
```

Comment:
See above, don't think this is the best place to do this validation. Rather than getting invalid creation I think we should catch the request and not create the model in the first place. This validator is heavily coupled to the question model and that's a bad smell for me.